.github/templates/macos_binary_build_workflow.yml.j2

```diff
@@ -40,6 +40,12 @@ on:
 {%- endfor %}
   workflow_dispatch:
 
+{%- if branches == "nightly" %}
+permissions:
+  id-token: write
+  contents: read
+{%- endif %}
```

Comment:
Same as above, this makes token available for all jobs in the workflow, but it's needed only for workflow job